python/cudf/cudf/core/series.py

```diff
@@ -4617,8 +4617,73 @@ def _get_dt_field(self, field):
             data=out_column, index=self.series._index, name=self.series.name
         )
 
-    def ceil(self, field):
-        out_column = self.series._column.ceil(field)
+    def ceil(self, freq):
+        """
+        Perform ceil operation on the data to the specified freq.
+
+        Parameters
+        ----------
+        freq : str
+            One of ["D", "H", "T", "min", "S", "L", "ms", "U", "us", "N"]
+            Must be a fixed frequency like 'S' (second) not 'ME' (month end).
+            See `frequency aliases <https://pandas.pydata.org/docs/\
+                user_guide/timeseries.html#timeseries-offset-aliases>`_
+            for more details on these aliases.
+
+        Returns
+        -------
+        Series
+            Series with all timestamps rounded up to the specified frequency.
+            The index is preserved.
+
+        Examples
+        --------
+        >>> import cudf
+        >>> t = cudf.Series(["2001-01-01 00:04:45", "2001-01-01 00:04:58",
+        ... "2001-01-01 00:05:04"], dtype="datetime64[ns]")
+        >>> t.dt.ceil("T")
+        0   2001-01-01 00:05:00
+        1   2001-01-01 00:05:00
+        2   2001-01-01 00:06:00
+        dtype: datetime64[ns]
+        """
+        out_column = self.series._column.ceil(freq)
+
+        return Series(
+            data=out_column, index=self.series._index, name=self.series.name
+        )
```

Comment:
```suggestion
        return Series._from_data(
            data=out_column, index=self.series._index, name=self.series.name
        )
```

Reply:
I think `data=out_column` should be `data={None: out_column}` or maybe better:
```python
Series._from_data(
            data={self.series.name: out_column}, index=self.series._index
        )
```
This might have to do with the failure below.

Reply:
Whoops thanks for the catch @isVoid .

Reply:
Oh right, thanks @isVoid and @vyasr! This makes sense now, when it said it couldn't iterate on a `DatetimeColumn`, I think that's what `_from_data` does (and hence we need to pass in something we can iterate on, eg. `{self.series.name: out_column}`